warcio/recordloader.py

```diff
@@ -126,22 +121,46 @@ def parse_record_stream(self, stream,
         # limit stream to the length for all valid records
         if length is not None and length >= 0:
             stream = LimitReader.wrap_stream(stream, length)
-
+            stream, is_verifying = self.wrap_digest_verifying_stream(stream, rec_type,
```

Comment:
If `DigestVerifyingReader` were to (and can) extend `LimitReader` and per my comment on returning the original stream if `check_digests` is false.

Then this could be simplified to 
```suggestion
            if check_digests:
                stream, is_verifying = self.wrap_digest_verifying_stream(stream, rec_type, rec_headers, length=length)
            else:
                stream, is_verifying = LimitReader.wrap_stream(stream, length), False
```

Reply:
Per the comment below, I think this is a good approach. If check_digests==False, there is no need to create a DigestVerifyingReader at all.